controls/listview/listview-features-load-on-demand.md

```diff
@@ -6,121 +6,118 @@ position: 8
+* [Use the **LoadOnDemand UserCommand** and add the loaded data to the source](#using-loadondemand-command).
 
-* Create a **ListViewLoadOnDemandCollection** instance as source and pass it to the ListView ItemsSource property.
-* Subscribe to **LoadOnDemand event** and add the loaded data to the source.
-* Use the **LoadOnDemand UserCommand** and add the loaded data to the source.
+>note All three approaches for loading items on demand in ListView work with both Automatic and Manual LoadOnDemandMode.
 
->note Either of the three approaches for loading items on demand in ListView works with both Automatic and Manual LoadOnDemandMode.
+### Using LoadOnDemandCollection
 
-## Using LoadOnDemandCollection
+To load items on demand, you can utilize the **ListViewLoadOnDemandCollection** and set it as ItemsSource for ListView. The ListViewLoadOnDemandCollection accepts an action in the constructor and this action is later executed by the ListView in a non-UI thread when new items are requested.
 
-One possible approach to load items on demand is by utilizing the **ListViewLoadOnDemandCollection** which should be set as ItemsSource of the ListView. The ListViewLoadOnDemandCollection accepts an action in the constructor and this action is later executed  by the ListView in a non-UI thread when new items are requested.
+The example below demonstrates how to use LoadOnDemandCollection:
 
-ListViewLoadOnDemandCollection can work both in **Manual** and **Automatic** mode. Below you can find an example with loading on demand through LoadOnDemandCollection.
+1. Define a sample ViewModel class with *Source* property of type ListViewLoadOnDemandCollection:
 
-First, let's define a sample ViewModel class with *Source* property of type ListViewLoadOnDemandCollection:
+    <snippet id='listview-loadondemand-loadondemandcollection-viewmodel'/>
 
-<snippet id='listview-loadondemand-loadondemandcollection-viewmodel'/>
+1. Define RadListView instance and bind its ItemsSource to the data in the viewmodel:
 
-Then, define RadListView instance and bind its ItemsSource to the data in the viewmodel:
+    <snippet id='listview-loadondemand-loadondemandcollection-declaration'/>
 
-<snippet id='listview-loadondemand-loadondemandcollection-declaration'/>
+1. Define the ListView namespace:
 
-You would also need to define the listview namespace:
+    <snippet id='xmlns-telerikdatacontrols'/>
 
-<snippet id='xmlns-telerikdatacontrols'/>
+1. Set the ViewModel as BindingContext in the page constructor:
 
-Finally, set the ViewModel as BindingContext in the page constructor:
+    <snippet id='listview-loadondemand-loadondemandcollection-binding'/>
 
-<snippet id='listview-loadondemand-loadondemandcollection-binding'/>
+### Using LoadOnDemand Event
 
-## Using LoadOnDemand Event
+Another way to handle loading more items is to use the **LoadOnDemand** event. This event is being raised on a UI thread, so in the event handler you can add items right away or asynchronously:
```

Comment:
The usage of _right away_ as opposed to _asynchronously_ is confusing for me. Does _right away_ mean synchronously? If not, is there another term that we can use?